CMakeLists.txt

```diff
@@ -0,0 +1,122 @@
+
+cmake_minimum_required(VERSION 3.4.3)
+
+list(APPEND CMAKE_MODULE_PATH
+     "${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules")
+
+project(XCTest
+        LANGUAGES
+          C)
+
+option(XCTEST_PATH_TO_LIBDISPATCH_SOURCE "Path to libdispatch source" "")
+option(XCTEST_PATH_TO_LIBDISPATCH_BUILD "Path to libdispatch build" "")
+
+option(XCTEST_PATH_TO_FOUNDATION_BUILD "Path to Foundation build" "")
+
+option(XCTEST_PATH_TO_COREFOUNDATION_BUILD "Path to CoreFoundation build" "")
+
+find_package(LLVM CONFIG)
+if(NOT LLVM_FOUND)
+  message(SEND_ERROR "Could not find LLVM; configure with -DCMAKE_PREFIX_PATH=/path/to/llvm/install")
```

Comment:
Question for you: my understanding is that there's also a method of configuration in which the `llvm-config` executable is invoked, is that correct? But I believe I heard that means of configuration is deprecated -- I'm not sure. Do you know the current situation is?

To be clear, I'm not asking you to change anything here, this looks right to me, I'm just wondering about `llvm-config`.

Reply:
This is the more modern, CMake way to handle this.  LLVM will install a CMake module that can be used.  If you use the `llvm-config` invocation approach, that requires the user to specify the path to the installation.  If you install LLVM to the system normally, this should JustWork(TM) so it is nicer.